src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = [];
+  const finalState = [];
+
+  newState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    for (const key in actions[i]) {
+      if (actions[i][key] === 'addProperties') {
+        Object.assign(newState, actions[i]['extraData']);
+      };
+
+      if (actions[i][key] === 'removeProperties') {
+        for (const key1 in actions[i]['keysToRemove']) {
```

Comment:
I think it needs to be forEach to avoid nesting loops inside if statement and/or make code more readable